### DISCUSSION TITLE
Redaction questions

### DISCUSSION THREAD

**Discussion by hjforbes** [opensafely/documentation#368] [Disclosure Control] [answered]
Anna has checked some output and come up with a couple of redaction queries:
- With histograms, it is sometimes hard to know whether some of the bins may have <=5 events. We can output the number of events in each bin as a check if needed.  However, it is difficult to see the exact number in each bin as the yscale is large, so we are not sure if this is necessary?
- Would an unadjusted HR based on events <=5 be considered disclosive? It may be possible to work out the exact number of events, so possibly safer to redact these? 

Thanks

**Comment by wjchulme** [comment, marked as answer]
For histograms, you have a few options:

* Just change all counts between 1 and 5 to 3, and explain clearly that this is what you've done. 
* Round counts, eg to the nearest 7
* Put a strip across the bottom to obscure counts between 1 and 5. This should only be done if the released image is a bitmap (eg png, jpeg) not a vector (eg svg) image, because with vectors the counts will be available within the raw code of the image file.

For any of these if the y-scale is large, then you won't notice the difference anyway in the image but it's necessary to avoid the problem with vector images containing the raw data.

I'm not sure what you've done with hazard ratios, but if you've based them on KM estimates of survival (or equivalently the raw cumulative survival probability if there is no censoring) then you can round the survival estimate / probability first, where the rounding resolution is equivalent to the redaction threshold (as it done here: https://github.com/opensafely/comparative-ve-research/blob/71e4786b9b9919f3beb3a914055280614143fab0/analysis/descriptive/km.R#L171) and then recalculate the HRs from that (ie, `hazard_{t} = -(survival_{t} - survival_{t-1}) / survival_{t-1}`). 

This assumes we're being strict about the "redact 5 or less" rule. At some point we may move to a "principles-based" system where if you can justify non-disclosivity despite low counts then you're ok to release. But I'm not sure we're there yet!

**Reply by hjforbes** [reply]
Super helpful!  Thanks Will
